src/site/content/en/reliable/codelab-background-sync/index.md

```diff
@@ -0,0 +1,212 @@
+---
+layout: codelab
+title: Building a resilient search experience with Workbox
+authors:
+  - demianrenzulli
+  - jeffposnick
+date: 2020-06-17
+description: |
+  How to implement a resilient search experience with Workbox, using  Background Sync and the Push API.
+glitch: resilient-search-codelab
+---
+
+This codelab shows you how to implement a resilient search experience with Workbox. The demo app it uses contains a search box that calls a server endpoint, and redirects the user to a basic HTML page.
+
+{% Aside %} 
+This codelab uses [Chrome DevTools](https://www.google.com/chrome/). Download Chrome if you don't already have it. 
+{% endAside %}
+
+## Measure
+
+Before adding optimizations, it's always a good idea to first analyze the current state of the application.
+
+{% Instruction 'remix' %}
+{% Instruction 'preview' %}
+
+Check how the website behaves when going offline:
+
+{% Instruction 'devtools-network', 'ol' %}
+1. Open Chrome DevTools and select the Network panel.
+1. In the [Throttling drop-down list](https://developers.google.com/web/tools/chrome-devtools/network/reference#throttling), select **Offline**.
+1. In the demo app enter a search query, and click on the search button.
+
+The standard browser error page is shown:
+
+![A screenshot of the default offline UX in the browser.](offline-ux-default.png)
+
+## Provide a fallback response
+
+The service worker contains the code to add the offline page to the [precache list](https://developers.google.com/web/tools/workbox/modules/workbox-precaching#explanation_of_the_precache_list), so it can always be cached at the service worker "install" event.
+
+Usually you would need to instruct Workbox to add this file to the precache list at build time, by integrating the library with your build tool of choice (e.g. [webpack](https://webpack.js.org/) or [gulp](https://gulpjs.com/)).
+
+For simplicity, we've already done it for you. The following code at `public/sw.js` does that:
+
+```javascript
+const FALLBACK_HTML_URL = ‘/index_offline.html’;
+…
+workbox.precaching.precacheAndRoute([FALLBACK_HTML_URL]);
+```
+
+{% Aside %} 
+To know more about ways of integrating Workbox with build tools, check the [webpack Workbox plugin](https://developers.google.com/web/tools/workbox/modules/workbox-webpack-plugin) and the [Gulp Workbox plugin](https://developers.google.com/web/tools/workbox/guides/codelabs/gulp).
+{% endAside %}
+
+Next, add the code below the `precacheAndRoute()` call in `public/sw.js`, to use the offline page as a fallback response:
+
+```javascript
+workbox.routing.setDefaultHandler(new workbox.strategies.NetworkOnly());
+
+workbox.routing.setCatchHandler(({event}) => {
+  switch (event.request.destination) {
+    case 'document':
+      return caches.match(FALLBACK_HTML_URL);
+      break;
+    default:
+      return Response.error();
+  }
+});
+```
+
+The code does the following:
```

Comment:
```suggestion

{% Aside %}
  The Glitch UI says `workbox is not defined` because it doesn't realize that the
  `importScripts()` call on line 1 is importing the library.
{% endAside %}

The code does the following:
```

![image](https://user-images.githubusercontent.com/4713486/85096073-11d4c400-b1a8-11ea-8b3d-417726177f70.png)